src/library/scala/concurrent/ExecutionContext.scala

```diff
@@ -166,6 +166,25 @@ object ExecutionContext {
     override final def reportFailure(t: Throwable): Unit = defaultReporter(t)
   }
 
+  /**
+   * This `ExecutionContext` is ideally suited to execute short-lived tasks on the `global` `ExecutionContext` as it
+   * attempts to locally batch the execution of nested tasks.
+   *
+   * WARNING: long-running and/or blocking tasks should be demarcated within `scala.concurrent.blocking`-blocks,
+   *          to ensure that any pending tasks in the current batch can be executed by another thread on `global`.
+   */
+  private[scala] object opportunistic extends ExecutionContextExecutor with BatchingExecutor {
```

Comment:
```suggestion
  private[scala] lazy val opportunistic: ExecutionContextExecutor = new ExecutionContextExecutor with BatchingExecutor {
```

Could we make it a lazy val? This would allow users to access it as follows:

```
implicit lazy val ec = (scala.concurrent.ExecutionContext: {def opportunistic: scala.concurrent.ExecutionContextExecutor}).opportunistic
```

I can't get this to work with it being an `object`.

Reply:
I mena, we could—alas it would be inconsistent w.r.t. the definition of `parasistic`.

Reply:
Also, you couldn't really obtain it like that since it would not be resilient to backwards incompatibilities—you'd have to fallback to global if `opportunistic` isn't available.

Reply:
> inconsistent w.r.t. the definition of `parasistic`

A lazy val would be consistent with the definition of `global` though :-)



> you couldn't really obtain it like that since it would not be resilient to backwards incompatibilities

Yes, but for end users that pull in the EC in their application this would be fine. The Scala version at compile and run time are the same.

Reply:
FWIW, @lrytz's suggestion would allow even Scala.js codebases to access it in the same way.

Reply:
>A lazy val would be consistent with the definition of global though :-)

Indeed! The only reason why `global` isn't an object is because it is dynamically configured & instantiated.

>Yes, but for end users that pull in the EC in their application this would be fine. The Scala version at compile and run time are the same.

Not necessarily true. A library might be built against 2.13.4 and be imported into a project which uses 2.13.2

Reply:
Yes, what I was trying to say: a library should *never* use the workaround to access the new field. But when writing an application, it's fine to use it in the application's own source code.

Reply:
Indeed—so we'd want to document both for libraries and for applications.